### ISSUE TITLE
don't know how to name or describe it, look at the log

### ISSUE THREAD

**Issue by admzzz** [Zackptg5/Cross-Compiled-Binaries-Android#9]
```
Cross Compiled Binaries v2.0(2)
by Zackptg5
===============================
BusyBox v1.31.1-Magisk
/sbin/.magisk/modules/ccbins/busybox
===============================
 
Testing internet connection [/] - OK

Fetching available binaries
 [|] - Done

Checking for mod updates
 [\][: 261: unknown operand
 [/][: 9: unknown operand
 [-][: 1: unknown operand
 [|][: 3: unknown operand
 [/] - Ccbins is up to date

Checking for binary updates
 [/]cat: read error: Is a directory
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\] - Updating /system/bin: Bash (5.16 -> )
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/] - Bash has been updated!
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Updating /system/bin: Bc (1.07.1 -> )
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Bc has been updated!
 [/]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\] - Updating /system/bin: Brotli (1.0.7 -> )
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-] - Brotli has been updated!
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Updating /system/bin: Coreutils (8.32 -> )
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Coreutils has been updated!
 [/]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/] - Updating /system/bin: Cpio (2.12 -> )
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Cpio has been updated!
 [/]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/] - Updating /system/bin: Curl (7.67.0 -> )
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Curl has been updated!
 [/]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-] - Updating /system/bin: Diffutils (3.7 -> )
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Diffutils has been updated!
 [/]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-] - Updating /system/bin: Findutils (4.7.0 -> )
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Findutils has been updated!
 [/]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/] - Updating /system/bin: Gawk (5.0.1 -> )
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-] - Gawk has been updated!
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\] - Updating /system/bin: Grep (3.4 -> )
 [|]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/] - Grep has been updated!
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Updating /system/bin: Gzip (1.10 -> )
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\] - Gzip has been updated!
 [/]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/] - Updating /system/bin: Htop (2.2.0 -> )
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/] - Htop has been updated!
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Updating /system/bin: Keycheck (1.0 -> )
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-] - Keycheck has been updated!
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\] - Updating /system/bin: Nano (4.9 -> )
 [|]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-] - Nano has been updated!
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Updating /system/bin: Patch (2.7.6 -> )
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/] - Patch has been updated!
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\] - Updating /system/bin: Patchelf (0.10 -> )
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Patchelf has been updated!
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-] - Updating /system/bin: Sed (4.8 -> )
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\] - Sed has been updated!
 [|]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Updating /system/bin: Strace (5.5 -> )
 [/]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Strace has been updated!
 [/]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-] - Updating /system/bin: Tar (1.32 -> )
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - Tar has been updated!
 [/]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/] - Updating /system/bin: Xmlstarlet (1.0 -> )
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\] - Xmlstarlet has been updated!
 [/]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/] - Updating /system/bin: Zip (3.0 -> )
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/] - Zip has been updated!
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\] - Updating /system/bin: Zsh (5.7.1 -> )
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/] - Zsh has been updated!
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\] - Updating /system/bin: Zstd (1.4.4 -> )
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\] - Zstd has been updated!
 [|] - Binaries are up to date

Checking for conflicting modules
rm: '/sbin/.magisk/modules/ccbins/tmp' is a directory
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [\]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [/]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [-]grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
grep: /sbin/.magisk/modules/ccbins/.binaries: No such file or directory
 [|] - The following conflicts were found:

cat: read error: Is a directory
 
===============================
Binary Installer ==============
===============================
Current install location:
/system/bin
===============================
Choose option =================
===============================
0: Refresh CCbins (remove everything - start fresh)
1: Change install location
2: List all installed binaries
3: List all downloadable binaries
4: Remove binary
5: Install binary
6: Backup current config
Type 'q' at any time to quit
--> 2
 
/system/bin/ccbins[505]: can't create /sbin/.magisk/modules/ccbins/tmp: Is a directory
===============================
/system/bin ===================
===============================
===============================
Binary Installer ==============
===============================
Current install location:
/system/bin
===============================
Choose option =================
===============================
0: Refresh CCbins (remove everything - start fresh)
1: Change install location
2: List all installed binaries
3: List all downloadable binaries
4: Remove binary
5: Install binary
6: Backup current config
Type 'q' at any time to quit
--> 
```

los 17.1, android 10
after this pseudo-update:
```
# htop
Error opening terminal: xterm-256color.
```
but htop works previously...

**Comment by diehard88** [comment]
that's lineage OS, right?

**Comment by admzzz** [comment]
yes, los 17.1

**Comment by Zackptg5** [comment]
I just released a new version of ccbins. Can you try completely uninstalling the old one (reboot and all), and then installing the new one from the repo?

**Comment by admzzz** [comment]
all ok (remove+reboot+install), the only problem (as it was in the previous version also) was the detecting of internet connection: it does not work in your script, when afwall is on, so after the installation with disabled afwall one had to repair safetynet check.

may i ask/suggest you three additions to your collection: nethogs, sqlite3 (i know that there is a module, but it usually contains old version) and midnight commander?

thank you for your work.

**Comment by Zackptg5** [comment]
This is the connection test function:
`test_connection() {
  (ping -q -c 1 -W 1 google.com >/dev/null 2>&1 & e_spinner "Testing internet connection") && echo " - OK" || { echo " - Error"; false; }
}`
Afwall shouldn't be blocking a ping to google so that'd be something you'd need to configure there
There was already a requst form nethogs but since I can't get libpcap to crosscompile, it's not happening sadly. I'll take a look at the other 2 when I get the chance

**Comment by Zackptg5** [comment]
Just added sqlite3 dynamic linked to testing repo, it'll be pushed to stable this week

**Comment by Zackptg5** [comment]
attempted midnight commander but ran into some compile issues so don't think that one's happening for now

**Comment by admzzz** [comment]
> This is the connection test function:
> `test_connection() { (ping -q -c 1 -W 1 google.com >/dev/null 2>&1 & e_spinner "Testing internet connection") && echo " - OK" || { echo " - Error"; false; } }`

hi, may i suppose another logic?
1) sophisticated user usually block in afwall all programs with root privileges;
2) magisk-manager is unblocked (in most cases), as it need to check for updates and to install 
 modules;
3) if ccbins script check magisk-manager's uuid and do su to this uuid, it would be passed out by afwall.

about sqlite static compiling:
`---Quote (Originally by adm_)---

---Quote---

was compiled using Android NDK and with a command line that is kinda hacked together...

---End Quote---

may i ask you to tell this command line?

---End Quote---

Sure if you think it'd be of any use.





Code:

/opt/android-ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86/bin/arm-linux-androideabi-gcc --sysroot /opt/android-ndk/platforms/android-9/arch-arm -I/home/user/Desktop/a/ncurses/include -I/home/user/Desktop/a/readline -fPIC -static -muclibc -fPIE -march=armv5te -mfloat-abi=softfp -mfpu=vfpv4-d16 -pipe -std=gnu99 -Wall -pedantic -O3 -I/home/user/Desktop/a/sqlite3 -I/home/user/Desktop/a -D_XOPEN_SOURCE=500 -I/system/usr/include/ncurses -I/system/usr/include -D_POSIX_C_SOURCE=1 -D_BSD_SOURCE=1 -D_GNU_SOURCE=1 -DSQLITE_THREADSAFE=0 -DSQLITE_OMIT_LOAD_EXTENSION=1 -USQLITE_CASE_SENSITIVE_LIKE -USQLITE_32BIT_ROWID -USQLITE_4_BYTE_ALIGNED_MALLOC -USQLITE_OMIT_EXPLAIN -DSQLITE_ENABLE_EXPLAIN_COMMENTS=1 -USQLITE_DEFAULT_AUTOVACUUM -USQLITE_ENABLE_FTS1 -USQLITE_ENABLE_FTS2 -DSQLITE_ENABLE_FTS3=1 -DSQLITE_ENABLE_FTS4=1 -USQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_FTS3_PARENTHESIS=1 -DSQLITE_ENABLE_FTS3_TOKENIZER=1 -USQLITE_DISABLE_FTS3_UNICODE -USQLITE_ENABLE_HIDDEN_COLUMNS -DSQLITE_

ENABLE_J

SON1=1 -DSQLITE_ENABLE_LOAD_EXTENSION=1 -DSQLITE_ENABLE_SNAPSHOT=1 -USQLITE_OMIT_LOCALTIME -DSQLITE_ENABLE_STAT4=1 -DSQLITE_DEFAULT_FOREIGN_KEYS=1 -DSQLITE_DEFAULT_MEMSTATUS=0 -DSQLITE_DEFAULT_WAL_SYNCHRONOUS=1 -DSQLITE_LIKE_DOESNT_MATCH_BLOBS=1 -DSQLITE_MAX_EXPR_DEPTH=0 -DSQLITE_OMIT_DECLTYPE=1 -DSQLITE_OMIT_DEPRECATED=1 -DSQLITE_OMIT_PROGRESS_CALLBACK=1 -DSQLITE_OMIT_SHARED_CACHE=1 -DSQLITE_USE_ALLOCA=1 -DSQLITE_TEMP_STORE=2 -DYYSTACKDEPTH=0 -DHAVE_LOCALTIME_R=1 -UHAVE_LOCALTIME_S -DHAVE_READLINE=1 -c sqlite3.c -o sqlite3.o

sqlite3.c: In function 'balance_nonroot':

sqlite3.c:71257:47: warning: array subscript is above array bounds [-Warray-bounds]

        pOld = iOld<nNew ? apNew[iOld] : apOld[iOld];

                                              ^

sqlite3.c:71257:47: warning: array subscript is above array bounds [-Warray-bounds]

sqlite3.c:71257:47: warning: array subscript is above array bounds [-Warray-bounds]

/opt/android-ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86/bin/arm-linux-androideabi-gcc --sysroot /opt/android-ndk/platforms/android-9/arch-arm -I/home/user/Desktop/a/ncurses/include -I/home/user/Desktop/a/readline -fPIC -static -muclibc -fPIE -march=armv5te -mfloat-abi=softfp -mfpu=vfpv4-d16 -pipe -std=gnu99 -Wall -pedantic -O3 -I/home/user/Desktop/a/sqlite3 -I/home/user/Desktop/a -D_XOPEN_SOURCE=500 -I/system/usr/include/ncurses -I/system/usr/include -D_POSIX_C_SOURCE=1 -D_BSD_SOURCE=1 -D_GNU_SOURCE=1 -DSQLITE_THREADSAFE=0 -DSQLITE_OMIT_LOAD_EXTENSION=1 -USQLITE_CASE_SENSITIVE_LIKE -USQLITE_32BIT_ROWID -USQLITE_4_BYTE_ALIGNED_MALLOC -USQLITE_OMIT_EXPLAIN -DSQLITE_ENABLE_EXPLAIN_COMMENTS=1 -USQLITE_DEFAULT_AUTOVACUUM -USQLITE_ENABLE_FTS1 -USQLITE_ENABLE_FTS2 -DSQLITE_ENABLE_FTS3=1 -DSQLITE_ENABLE_FTS4=1 -USQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_FTS3_PARENTHESIS=1 -DSQLITE_ENABLE_FTS3_TOKENIZER=1 -USQLITE_DISABLE_FTS3_UNICODE -USQLITE_ENABLE_HIDDEN_COLUMNS -DSQLITE_

ENABLE_J

SON1=1 -DSQLITE_ENABLE_LOAD_EXTENSION=1 -DSQLITE_ENABLE_SNAPSHOT=1 -USQLITE_OMIT_LOCALTIME -DSQLITE_ENABLE_STAT4=1 -DSQLITE_DEFAULT_FOREIGN_KEYS=1 -DSQLITE_DEFAULT_MEMSTATUS=0 -DSQLITE_DEFAULT_WAL_SYNCHRONOUS=1 -DSQLITE_LIKE_DOESNT_MATCH_BLOBS=1 -DSQLITE_MAX_EXPR_DEPTH=0 -DSQLITE_OMIT_DECLTYPE=1 -DSQLITE_OMIT_DEPRECATED=1 -DSQLITE_OMIT_PROGRESS_CALLBACK=1 -DSQLITE_OMIT_SHARED_CACHE=1 -DSQLITE_USE_ALLOCA=1 -DSQLITE_TEMP_STORE=2 -DYYSTACKDEPTH=0 -DHAVE_LOCALTIME_R=1 -UHAVE_LOCALTIME_S -DHAVE_READLINE=1 -c shell.c -o shell.o

/opt/android-ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86/bin/arm-linux-androideabi-gcc --sysroot /opt/android-ndk/platforms/android-9/arch-arm -L/home/user/Desktop/a/ncurses/lib -L/home/user/Desktop/a/readline -fPIC -static -muclibc -pie -L/home/user/Desktop/a/sqlite3 -L/home/user/Desktop/a -fno-strict-aliasing -Wl,--no-as-needed -Wl,--gc-sections -Wl,--dynamic-linker,/system/bin/linker -Wl,-z,nocopyreloc -Wl,-z,relro -Wl,-z,text -Wl,--no-omagic -Wl,--fix-cortex-a8 -s sqlite3.o shell.o -o sqlite3-armv5te -lm -lreadline -lhistory -lncurses

I had previously also compiled *ncurses* and *readline*, they're available in directories one up from SQLite's. They were compiled long time ago so I don't remember the exact procedure. I just keep them there and use them when necessary.`

**Comment by Zackptg5** [comment]
I actually already switched to github on the last update. I'm not going to change uuid of the commands in this, all of the network activity are done with github now and nothing else so they should just whitelist github or something

I already have static compiled ncurses so at least that much is done, I'll look into that more when I get the chance. Could you open a new issue with that quoted info in it? Thanks

**Comment by admzzz** [comment]
>  I'm not going to change uuid of the commands in this, all of the network activity are done with github now and nothing else so they should just whitelist github or something
all network activity are doing from the shell with uid of ccbins, which have root privileges and because of them are not passed through afwall. magisk-manager, on contrary, is whitelisted in afwall and would be passed through.
it is three line of code to check and use parent's uid (magisk is the parent of ccbins shell), maybe you'll agree to change your decision?

> Could you open a new issue with that quoted info in it? Thanks
Done
